src/pylast/__init__.py

```diff
@@ -906,68 +913,41 @@ def _download_response(self):
             self.network._delay_call()
 
         username = self.params.pop("username", None)
-        username = f"?username={username}" if username is not None else ""
-
-        data = []
-        for name in self.params.keys():
-            data.append("=".join((name, quote_plus(_string(self.params[name])))))
-        data = "&".join(data)
-
-        headers = {
-            "Content-type": "application/x-www-form-urlencoded",
-            "Accept-Charset": "utf-8",
-            "User-Agent": "pylast/" + __version__,
-        }
+        username = "" if username is None else f"?username={username}"
 
         (host_name, host_subdir) = self.network.ws_server
 
         if self.network.is_proxy_enabled():
-            conn = HTTPSConnection(
-                context=SSL_CONTEXT,
-                host=self.network._get_proxy()[0],
-                port=self.network._get_proxy()[1],
+            client = httpx.Client(
+                verify=SSL_CONTEXT,
+                base_url=f"https://{host_name}",
+                headers=HEADERS,
+                proxies=f"http://{self.network._get_proxy()}",
```

Comment:
I recommend allowing arbitrary values for `proxies` to be passed through directly. As per the documentation, there are many different possible values for this and it would be folly to attempt to support all of them up front, or to artificially restrict them like has been done here.

https://www.python-httpx.org/advanced/#http-proxying

Reply:
Like this?

```suggestion
                proxies=self.network._get_proxy(),
```

And this?

```diff
-    def enable_proxy(self, host, port):
+    def enable_proxy(self, proxy):
         """Enable a default web proxy"""

-        self.proxy = [host, _number(port)]
+        self.proxy = proxy
         self.proxy_enabled = True
```

Reply:
Yep, that makes sense to me :+1: